app/bundles/MessengerBundle/Message/PageHitNotification.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\MessengerBundle\Message;
+
+use Mautic\MessengerBundle\Message\Traits\MessageRequestTrait;
+use Symfony\Component\HttpFoundation\Request;
+
+class PageHitNotification
+{
+    use MessageRequestTrait;
+
+    /**
+     * @var array<string,mixed>
+     */
+    private array $request;
+
+    /**
+     * @param array<string,mixed>|Request $request
+     */
+    public function __construct(
+        private int $hitId,
+        private ?int $pageId,
```

Comment:
but it is required, there is no default value, it just may be null, same as the request, it can have two values but must have a value.

Reply:
A nullable param means it is not required. `null` means unknown, right? With this order, you are forcing users of this class to instantiate the object like this:
```php
new PageHitNotification(1, null, [], null, true, true);
```
If the nullable attributes would be optional at the end like this:
```php
public function __construct(
        private int $hitId,
        array|Request $request,
        private bool $isNew,
        private bool $isRedirect,
        private ?int $pageId = null,
        private ?int $leadId = null,
        \DateTimeInterface $eventTime = null
```
then it would look like this:
```php
new PageHitNotification(1, [], true, true);
```
And you would get the same values into the object.

This is very minor request. If you want this particular order. I'm not going to block this PR because of it.

Reply:
I just like to understand and learn. I will change it.